docs/pipelines/overview/configuration.md

```diff
@@ -1,95 +1,368 @@
 
-A sample `config.yml` with a full set of options is below.
+If you are currently using YAML configurations, you should not migrate to the new configuration system at this time.
+:::
+
+Pipelines relies on configurations written in [HashiCorp Configuration Language (HCL)](https://github.com/hashicorp/hcl) to drive dynamic behavior. These configurations are primarily used by Pipelines to determine how to interact with cloud environments within the context of the Infrastructure As Code (IAC) within a code repository.
+
+At a high level, Pipelines will read these configurations by parsing all files that end with `.hcl` within a directory named `.gruntwork` or a single file named `gruntwork.hcl`. In typical usage, the configurations that are global to the IAC in a repository will be defined within a `.gruntwork` directory at the root of the repository, and configurations that are specific to a particular `terragrunt.hcl` file (a unit) located in the same directory as the `terragrunt.hcl` file.
+
+## Minimum Required Configuration
+
+The minimum configurations required for Pipelines to operate correctly will vary depending on context. For the most common usage, Pipelines will need to be able to determine how to authenticate with a cloud provider in order to run Terragrunt commands. If it is not able to determine how to authenticate within a context where it needs to be able to do so, Pipelines will throw an error.
+
+The following is an example of a minimal configuration that would allow Pipelines to determine how to authenticate with AWS for exactly one unit of IAC:
+
+```hcl
+# gruntwork.hcl
+unit {
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+Placing this configuration in a `gruntwork.hcl` file in the same directory as a `terragrunt.hcl` file will cause Pipelines to assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands, using [OIDC](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) to authenticate to AWS and assume that role.
+
+In most circumstances, the same role would be assumed by multiple units of configuration within a repository (e.g. all units within a given directory configure resources for the same AWS account). In that circumstance, it would be more appropriate not to define a `gruntwork.hcl` file at all, and instead define a `.gruntwork` directory at the root of the repository with a file ending in `.hcl` that contains configurations that are applicable to all units within that directory.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
```

Comment:
I like that you introduce `gruntwork.hcl` in a single folder first, then walk through and expand to the `filter` usecase in teh root of the repo, nice.